src/server/game/Guilds/Guild.cpp

```diff
@@ -1672,6 +1672,44 @@ void Guild::_SendSetNewGuildMaster(Member const* guildMaster, Member const* newG
     BroadcastPacket(&data);
 }
 
+void Guild::_SendRemovePlayerFromGuild(ObjectGuid removedGuid, std::string const& removedName, Player const* remover) const
+{
+    ObjectGuid removerGuid = remover ? remover->GetGUID() : 0;
+    WorldPacket data(SMSG_GUILD_LEAVE, 8 + 4 + 1 + 1 + removedName.length() + (remover ? (8 + 4 + 1 + 1 + remover->GetName().length()) : 0));
+    data.WriteGuidMask(removedGuid, 2);
+    data.WriteBits(removedName.length(), 6);
+    data.WriteGuidMask(removedGuid, 6, 5);
+    data.WriteBit(remover != nullptr);
+
+    if (remover)
+    {
+        data.WriteBit(0); // unk bool
+        data.WriteBit(0); // unk bool
```

Comment:
First bool denotes the inclusion of the kickers name
Second bool is fake as it's value at the client is written in to the same structure offset as the first guid bit

Reply:
Thanks added comments there for the structure.